lecture/src/main/java/de/tum/in/www1/artemis/lecture/web/rest/ExerciseUnitResource.java

```diff
@@ -0,0 +1,117 @@
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+
+import static de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException.NOT_ALLOWED;
+
+@RestController
+@RequestMapping("api/")
+public class ExerciseUnitResource {
+
+    private final Logger log = LoggerFactory.getLogger(ExerciseUnitResource.class);
+
+    private static final String ENTITY_NAME = "exerciseUnit";
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final ExerciseUnitRepository exerciseUnitRepository;
+
+    private final LectureRepository lectureRepository;
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    public ExerciseUnitResource(LectureRepository lectureRepository, ExerciseUnitRepository exerciseUnitRepository, AuthorizationCheckService authorizationCheckService) {
+        this.exerciseUnitRepository = exerciseUnitRepository;
+        this.lectureRepository = lectureRepository;
+        this.authorizationCheckService = authorizationCheckService;
+    }
+
+    /**
+     * POST lectures/:lectureId/exercise-units : creates a new exercise unit.
+     *
+     * @param lectureId    the id of the lecture to which the attachment unit should be added
+     * @param exerciseUnit the exercise unit that should be created
+     * @return the ResponseEntity with status 201 (Created) and with body the new exercise unit
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("lectures/{lectureId}/exercise-units")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<ExerciseUnit> createExerciseUnit(@PathVariable Long lectureId, @RequestBody ExerciseUnit exerciseUnit) throws URISyntaxException {
+        log.debug("REST request to create ExerciseUnit : {}", exerciseUnit);
+        if (exerciseUnit.getId() != null) {
+            throw new BadRequestAlertException("An exercise unit must have an ID to be updated", ENTITY_NAME, "idnull");
+        }
+        Optional<Lecture> lectureOptional = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoals(lectureId);
+        if (lectureOptional.isEmpty()) {
+            throw new BadRequestAlertException("Invalid id", ENTITY_NAME, "not found");
+        }
+        Lecture lecture = lectureOptional.get();
+        if (lecture.getCourse() == null) {
+            throw new ConflictException();
+        }
+        if (!authorizationCheckService.isAtLeastEditorInCourse(lecture.getCourse(), null)) {
+            throw new AccessForbiddenException(NOT_ALLOWED);
+        }
```

Comment:
```suggestion
        authorizationCheckService.hasAtLeastRoleInCourseElseThrow(Role.EDITOR, lecture.getCourse(), null));
```
The authorisation checks here and in the other endpoint could be simplified, as no special message is put into the `AccessForbiddenException`.

Reply:
This probably can be done in the other Resource classes as well